### ISSUE TITLE
GetRankingInfo - Typecasting error

### ISSUE THREAD

**Issue by blok5025** [algolia/algoliasearch-client-dart#6]
Looking at the open issues, there seems to be more than one situation where typecasting and null checks are not setup properly. This one was caused by  ```getRankingInfo: true``` inside of ```SearchForHits()```.

Here is an example to reproduce:
```
final clothing = ['Shirts', 'Pants'];
  final size = [2, 8];
  final filters =
      'type:"${clothing.join('" OR type:"')}" AND size:${sizeRange[0]} TO ${sizeRange[1]}';

  final queryHits = SearchForHits(
    indexName: 'shops',
    aroundLatLng: latLng,
    aroundRadius: radiusInt,
    getRankingInfo: true,
    filters: filters,
    hitsPerPage: 20,
  );
  
  final responseHits = await client.searchIndex(request: queryHits);
```

Error Log:
```
E/flutter (15946): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: CheckedFromJsonException
E/flutter (15946): Could not create `RankingInfo`.
E/flutter (15946): There is a problem with "promoted".
E/flutter (15946): type 'Null' is not a subtype of type 'bool' in type cast
E/flutter (15946): #0      $checkedConvert (package:json_annotation/src/checked_helpers.dart:91:5)
E/flutter (15946): #1      $checkedCreate.checkedConvert (package:json_annotation/src/checked_helpers.dart:28:7)
E/flutter (15946): #2      _$RankingInfoFromJson.<anonymous closure> (package:algolia_client_search/src/model/ranking_info.g.dart:31:36)
E/flutter (15946): #3      $checkedCreate.<anonymous closure> (package:json_annotation/src/checked_helpers.dart:33:22)
E/flutter (15946): #4      $checkedNew (package:json_annotation/src/checked_helpers.dart:51:23)
E/flutter (15946): #5      $checkedCreate (package:json_annotation/src/checked_helpers.dart:30:10)
E/flutter (15946): #6      _$RankingInfoFromJson (package:algolia_client_search/src/model/ranking_info.g.dart:9:65)
E/flutter (15946): #7      new RankingInfo.fromJson (package:algolia_client_search/src/model/ranking_info.dart:114:7)
E/flutter (15946): #8      _$HitFromJson.<anonymous closure>.<anonymous closure> (package:algolia_client_search/src/model/hit.g.dart:23:33)
E/flutter (15946): #9      $checkedConvert (package:json_annotation/src/checked_helpers.dart:87:20)
E/flutter (15946): #10     $checkedCreate.checkedConvert (package:json_annotation/src/checked_helpers.dart:28:7)
E/flutter (15946): #11     _$HitFromJson.<anonymous closure> (package:algolia_client_search/src/model/hit.g.dart:19:39)
E/flutter (15946): #12     $checkedCreate.<anonymous closure> (package:json_annotation/src/checked_helpers.dart:33:22)
E/flutter (15946): #13     $checkedNew (package:json_annotation/src/checked_helpers.dart:51:23)
E/flutter (15946): #14     $checkedCreate (package:json_annotation/src/checked_helpers.dart:30:10)
E/flutter (15946): #15     _$HitFromJson (package:algolia_client_search/src/model/hit.g.dart:9:49)
E/flutter (15946): #16     new Hit.fromJson (package:algolia_client_search/src/model/hit.dart:61:22)
E/flutter (15946): #17     _$SearchResponseFromJson.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:algolia_client_search/src/model/search_response.g.dart:71:35)
E/flutter (15946): #18     MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
E/flutter (15946): #19     ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
E/flutter (15946): #20     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
E/flutter (15946): #21     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
E/flutter (15946): #22     new List.of (dart:core-patch/array_patch.dart:39:18)
E/flutter (15946): #23     ListIterable.toList (dart:_internal/iterable.dart:224:7)
E/flutter (15946): #24     _$SearchResponseFromJson.<anonymous closure>.<anonymous closure> (package:algolia_client_search/src/model/search_response.g.dart:72:20)
E/flutter (15946): #25     $checkedConvert (package:json_annotation/src/checked_helpers.dart:87:20)
E/flutter (15946): #26     $checkedCreate.checkedConvert (package:json_annotation/src/checked_helpers.dart:28:7)
E/flutter (15946): #27     _$SearchResponseFromJson.<anonymous closure> (package:algolia_client_search/src/model/search_response.g.dart:68:32)
E/flutter (15946): #28     $checkedCreate.<anonymous closure> (package:json_annotation/src/checked_helpers.dart:33:22)
E/flutter (15946): #29     $checkedNew (package:json_annotation/src/checked_helpers.dart:51:23)
E/flutter (15946): #30     $checkedCreate (package:json_annotation/src/checked_helpers.dart:30:10)
E/flutter (15946): #31     _$SearchResponseFromJson (package:algolia_client_search/src/model/search_response.g.dart:10:5)
E/flutter (15946): #32     new SearchResponse.fromJson (package:algolia_client_search/src/model/search_response.dart:234:7)
E/flutter (15946): #33     SearchClientExt.searchIndex (package:algolia_client_search/src/extension/search.dart:13:27)
E/flutter (15946): <asynchronous suspension>
```

**Comment by peaklev** [comment]
I am seeing the same thing. This issue makes it impossible to query locations and get back results with the distance to each location.

**Comment by AbdulBasitab** [comment]
I am also facing the same issue as well while querying for location based search

**Comment by jord-an** [comment]
This appears to stem from Algolia not including a `promoted` attribute in its response to a query. 

Although untested, this should be as simple as changing:

`required this.promoted` to `this.promoted`
and 
`final bool promoted` to `final bool? promoted`

in `/lib/src/model/ranking_info.dart` which will allow it to accept a Null value.

I ran into a similar issue using the [algolia_helper_flutter](https://github.com/algolia/algoliasearch-helper-flutter) package which doesn't include the ability to specify `getRankingInfo`. I have included this function plus a workaround for the missing promoted attribute into a fork of this package here on GitHub if that's of interest. It can be found [here](https://github.com/jord-an/algoliasearch-helper-flutter).

I haven't added my algolia_helper_flutter changes as a pull request as the workaround should not really be included, instead this git should be altered to correctly mimic the current API responses.
